app/ts/library/provider.ts

```diff
@@ -1,38 +1,40 @@
 import { batch, Signal } from '@preact/signals'
-import { providers, Signer, utils, Wallet } from 'ethers'
+import { Block, BrowserProvider, getAddress, HDNodeWallet, Wallet } from 'ethers'
 import { MEV_RELAY_GOERLI, MEV_RELAY_MAINNET } from '../constants.js'
-import { AppSettings, Signers } from '../types/types.js'
+import { AddressParser, EthereumAddress } from '../types/ethereumTypes.js'
+import { AppSettings, BlockInfo, Signers } from '../types/types.js'
 
 export type ProviderStore = {
-	provider: providers.Web3Provider
-	authSigner: Signer,
+	provider: BrowserProvider
 	_clearEvents: () => unknown
-	walletAddress: string
-	chainId: number
+	authSigner: HDNodeWallet,
+	walletAddress: EthereumAddress
+	chainId: bigint
 }
 
 const addProvider = async (
 	store: Signal<ProviderStore | undefined>,
-	appSettings: Signal<AppSettings>,
-	provider: providers.Web3Provider,
+	provider: BrowserProvider,
 	clearEvents: () => unknown,
+	appSettings: Signal<AppSettings>
 ) => {
 	const [signer, network] = await Promise.all([provider.getSigner(), provider.getNetwork()])
 	const address = await signer.getAddress()
 	if (store.peek()) removeProvider(store)
 
-	if (![1, 5].includes(network.chainId)) {
+	const parsedAddress = AddressParser.parse(getAddress(address))
+	if (!parsedAddress.success) throw new Error("Provider provided invalid address!")
+
+	if (![1n, 5n].includes(network.chainId)) {
```

Comment:
We could make separate const for these networks as these things seem to be repeated in multiple places. You could then join chain Id, chain name and provider relay in a single variable